client/my-sites/upgrades/checkout/domain-details-form.jsx

```diff
@@ -318,9 +319,13 @@ export default React.createClass( {
 	},
 
 	recordSubmit() {
+		const errors = formState.getErrorMessages( this.state.form );
 		analytics.tracks.recordEvent( 'calypso_contact_information_form_submit', {
-			errors: formState.getErrorMessages( this.state.form )
+			errors,
+			errors_count: errors && errors.length || 0,
+			submission_count: this.state.submissionCount + 1
 		} );
+		this.setState( { submissionCount: this.state.submissionCount + 1 } );
```

Comment:
Maybe you could set the state first and then just use `this.state.submissionCount` when recording the event? That way you only have to remember to increment the count in one place.